python/unit.cpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+
+#include "unit.h"
+
+#include "scipp/units/string.h"
+
+#include "dtype.h"
+
+using namespace scipp;
+namespace py = pybind11;
+
+std::tuple<units::Unit, int64_t, int64_t>
+get_time_unit(const std::optional<scipp::units::Unit> value_unit,
+              const std::optional<scipp::units::Unit> variance_unit,
+              const std::optional<scipp::units::Unit> dtype_unit,
+              const units::Unit sc_unit) {
+  // TODO proper dimension check
```

Comment:
What is not proper here, can you explain better?

Reply:
Only detects the 3 listed time units. With the new dynamic handling, e.g. `ms` would not be recognised correctly.